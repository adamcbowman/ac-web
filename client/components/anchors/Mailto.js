import React from 'react'
import PropTypes from 'prop-types'
import { INFO } from 'constants/emails'
import { clean } from 'utils/object'

Mailto.propTypes = {
    email: PropTypes.string,
    title: PropTypes.string,
    subject: PropTypes.string,
    cc: PropTypes.string,
    bcc: PropTypes.string,
    body: PropTypes.string,
    children: PropTypes.element.isRequired,
}

export default function Mailto({
    email = INFO,
    title = email,
    subject,
    cc,
    bcc,
    body,
    children,
    ...rest
}) {
    const params = new URLSearchParams(clean({ subject, cc, bcc, body }))
    const search = params.toString()
    let href = `mailto:${email}`

    if (search) {
        href = `${href}?${search}`
    }

    return (
        <a href={href} title={title} {...rest}>
            {children || email}
        </a>
    )
}
