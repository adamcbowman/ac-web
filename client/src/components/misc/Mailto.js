import React, {PropTypes, DOM} from 'react'
import {compose, defaultProps, withProps, setDisplayName, setPropTypes} from 'recompose'
import Url from 'url'
import {AVALANCHECANADA} from 'constants/emails'

export default compose(
    setDisplayName('Mailto'),
    setPropTypes({
        email: PropTypes.string,
        title: PropTypes.string,
        subject: PropTypes.string,
        cc: PropTypes.string,
        bcc: PropTypes.string,
        body: PropTypes.string,
    }),
    defaultProps({
        email: AVALANCHECANADA,
        title: 'Email Avalanche Canada',
    }),
    withProps(({email, title, children, subject, cc, bcc, body}) => ({
        children: children || email,
        href: Url.format({
            protocol: 'mailto',
            pathname: email,
            query: {
                subject,
                cc,
                bcc,
                body
            },
        }),
    }))
)(DOM.a)
