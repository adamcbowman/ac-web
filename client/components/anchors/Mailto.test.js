import React from 'react'
import Renderer from 'react-test-renderer'
import Mailto from './Mailto'

test('mailto component', () => {
    let mailto = Renderer.create(<Mailto email="kguillotte@avalanche.ca" />)

    expect(mailto).toMatchSnapshot()

    mailto = Renderer.create(
        <Mailto email="kguillotte@avalanche.ca">Karl Guillotte</Mailto>
    )

    expect(mailto).toMatchSnapshot()
})
