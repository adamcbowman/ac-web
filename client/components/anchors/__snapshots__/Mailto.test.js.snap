// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mailto component 1`] = `
<a
  href="mailto:kguillotte@avalanche.ca"
  title="kguillotte@avalanche.ca"
>
  kguillotte@avalanche.ca
</a>
`;

exports[`mailto component 2`] = `
<a
  href="mailto:kguillotte@avalanche.ca"
  title="kguillotte@avalanche.ca"
>
  Karl Guillotte
</a>
`;
